ci/.gitlab-ci.yml

```diff
@@ -34,7 +34,7 @@ cpu debug build:
     IMAGE: $CI_REGISTRY_IMAGE/debug/deploy2:$CI_COMMIT_SHA
   script:
     - BUILD_IMAGE=$BUILD_IMAGE_BASE`sha256sum $BUILD_DOCKER_FILE | head -c 64`
-    - docker pull $BUILD_IMAGE || docker build -t $BUILD_IMAGE -f $BUILD_DOCKER_FILE --network=host .
+    - docker build -t $BUILD_IMAGE --cache-from $BUILD_IMAGE --build-arg BUILDKIT_INLINE_CACHE=1 -f ci/docker/debug/build.Dockerfile --network=host .
```

Comment:
please use $BUILD_DOCKER_FILE to be sure that the same filename is used to compute the hash and to build the container